src/v/cluster/tests/partition_allocator_tests.cc

```diff
@@ -533,9 +567,15 @@ FIXTURE_TEST(rack_aware_assignment_1, partition_allocator_fixture) {
         nodes.insert(node_id);
     }
     BOOST_REQUIRE(nodes.size() == 3);
-    BOOST_REQUIRE(nodes.contains(model::node_id(0))); // rack-a
-    BOOST_REQUIRE(nodes.contains(model::node_id(2))); // rack-b
-    BOOST_REQUIRE(nodes.contains(model::node_id(4))); // rack-c
+    BOOST_REQUIRE(
+      nodes.contains(model::node_id(0))
+      || nodes.contains(model::node_id(1))); // rack-a
+    BOOST_REQUIRE(
+      nodes.contains(model::node_id(2))
+      || nodes.contains(model::node_id(3))); // rack-b
+    BOOST_REQUIRE(
+      nodes.contains(model::node_id(4))
+      || nodes.contains(model::node_id(5))); // rack-c
```

Comment:
This existing test relied on exactly which node would be chosen when ties occur: we loosen the check so that either valid node (out of the two on the rack) is fine. That is, the check is "is one node from each rack chosen").